File: 2021-09-02/JannaKim.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<string> split(string line){
    vector<string> words;
    size_t pos = 0;
    while( (pos = line.find(' ')) != string::npos) {
        words.push_back(line.substr(0, pos));
        line.erase(0, pos + 1);
    }
    words.push_back(line.substr(0, pos));

    return words;
}

string solution(vector<string> table, vector<string> languages, vector<int> preference) {

    string ans = "";
    int mx = -1;

    for(string &column : table){
        vector<string> col = split(column);
        int score = 0;
        for(int i = 1; i < 1 + 5; ++i){
            for(int j = 0; j < languages.size(); ++j){
                if(col[i] == languages[j]) score += (6 - i) * preference[j];
            }
        }
        if(score > mx) {
            mx = score;
            ans = col[0];
        }
        else if(score == mx){
            if(ans.compare(col[0]) > 0) ans = col[0];
        }
    }
    
    return ans;
}

File: 2021-09-02/JannaKim.py
def solution(table, languages, preference):
    occupation = {}
    
    for line in table:
        line = line.split()
        occupation[ line[0] ] = line[1:]
        
    ans = ""
    mx = -1
    
    for k, v in occupation.items():
        score = 0
        for lang, pref in zip(languages, preference):
            for i in range(5):
                if v[i] == lang:
                    score += (5 - i) * pref
                    break
                    
        if score > mx:
            mx = score
            ans = k
            
        elif score == mx:
            if ans > k:
                ans = k

    return ans